### ISSUE TITLE
Regression in support for properties beginning with "is"

### ISSUE THREAD

**Issue by ellisjoe** [immutables/immutables#328]
Similar to the issue here: https://github.com/immutables/immutables/issues/239

except properties that don't include the @JsonProperty annotation aren't properly deserialized.

Test:
```
@Value.Immutable
@JsonSerialize(as = ImmutableTestClass.class)
@JsonDeserialize(as = ImmutableTestClass.class)
public abstract class TestClass {

    public abstract boolean isTest();

    public static void main(String[] args) throws IOException {
        TestClass testClass = ImmutableTestClass.builder().isTest(true).build();
        ObjectMapper objectMapper = new ObjectMapper()
                .registerModule(new GuavaModule());

        objectMapper.readValue(objectMapper.writeValueAsString(testClass), TestClass.class);
    }

}
```

Exception:
```
Exception in thread "main" com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "test" (class com.ImmutableTestClass$Json), not marked as ignorable (one known property: "isTest"])
 at [Source: {"test":true}; line: 1, column: 13] (through reference chain: com.Json["test"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:51)
	at com.fasterxml.jackson.databind.DeserializationContext.reportUnknownProperty(DeserializationContext.java:817)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:958)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1324)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1302)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:249)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:169)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:144)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1069)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:295)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:142)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3562)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2578)
	at com.TestClass.main(TestClass.java:26)
```

**Comment by ldriscoll** [comment]
don't you have to add `get = {"is*", "get*"}` to the style?

**Comment by ellisjoe** [comment]
any reason this worked prior to 2.1?

**Comment by ldriscoll** [comment]
There were a lot of changes around handling jackson during the 2.1.x phase, where where intermediary objects where added; it's possible that that is what you're seeing.  

I think if you don't declare `is` as a getter then immutables will only deserialise `isTest` to your field; however jackson's default is to serialise `isXy` and `getXy` as `xy`; and since immutables thinks that everything that is not `getX` should be serialised and deserialised as the field name, there is confusion.

**Comment by ellisjoe** [comment]
So it's expected behavior that I can create an Immutable object that doesn't serialize and deserialize without a special style parameter added?

**Comment by elucash** [comment]
Please, I appreciate that any kind of incompatibility may be frustrating. But round trip to and from JSON should work correctly anyway, so here we definitely have a bug, regardless of expectation about "is*" support (which will still be opt-in using styles).

**Comment by elucash** [comment]
Roundtrip serialization and deserialization should work as expected. By default `is` is not autotrimmed, but this may be configured using styles
Released in 2.2.1 Thank you!
